packages/backend/src/app/flows/flow-run/flow-run-service.ts

```diff
@@ -79,27 +107,49 @@ export const flowRunService = {
         return flowRun
     },
 
-    async start({ flowVersionId, payload, environment }: StartParams): Promise<FlowRun> {
+    async pause(params: PauseParams): Promise<void> {
+        logger.warn(`[FlowRunService#pause] flowRunId=${params.flowRunId} pauseType=${params.pauseMetadata.step}`)
+
+        logger.debug(params, '[FlowRunService#pause] params')
+
+        const { flowRunId, pauseMetadata } = params
+
+        await repo.update(flowRunId, {
+            status: ExecutionOutputStatus.PAUSED,
+            pauseMetadata: pauseMetadata as QueryDeepPartialEntity<PauseMetadata>,
+        })
+
+        const flowRun = await repo.findOneByOrFail({ id: flowRunId })
+
+        await flowRunSideEffects.pause({ flowRun })
+    },
+
+    async start({ projectId, flowVersionId, flowRunId, payload, environment, executionType }: StartParams): Promise<FlowRun> {
+        logger.warn(`[flowRunService#start]  flowVersionId=${flowVersionId} flowRunId=${flowRunId} executionType=${executionType}`)
         logger.info(`[flowRunService#start]  flowVersionId=${flowVersionId}`)
 
         const flowVersion = await flowVersionService.getOneOrThrow(flowVersionId)
-        const flow = (await flowRepo.findOneBy({ id: flowVersion.flowId }))!
+
+        const flow = await flowService.getOneOrThrow({
```

Comment:
I think this is not needed, and there is two log statement

Reply:
what is not needed? can you please explain?